apisix/consumer.lua

```diff
@@ -186,24 +204,43 @@ function _M.consumers()
 end
 
 
-local function create_consume_cache(consumers_conf, key_attr)
+local create_consume_cache
+do
+    local consumers_plugin_key_lrucache_tab = {}
+
+local function create_new_consumer(consumer)
+    local new_consumer = core.table.clone(consumer)
+    new_consumer.auth_conf = secret.fetch_secrets(new_consumer.auth_conf, false)
+    return new_consumer
+end
+
+
+function create_consume_cache(consumers_conf, key_attr, plugin_name)
     local consumer_names = {}
+    local lru_cache = consumers_plugin_key_lrucache_tab[plugin_name]
+    if lru_cache == nil then
+        lru_cache = core.lrucache.new({
+            ttl = 60 * 60 * 24, count = 20480
+        })
+        consumers_plugin_key_lrucache_tab[plugin_name] = lru_cache
+    end
 
     for _, consumer in ipairs(consumers_conf.nodes) do
         core.log.info("consumer node: ", core.json.delay_encode(consumer))
-        local new_consumer = core.table.clone(consumer)
-        new_consumer.auth_conf = secret.fetch_secrets(new_consumer.auth_conf, true,
-                                                        new_consumer.auth_conf, "")
-        consumer_names[new_consumer.auth_conf[key_attr]] = new_consumer
+        local new_consumer = lru_cache(consumer.auth_conf[key_attr],
```

Comment:
I think: this should be enough, and it seems easier

```suggestion
        local new_consumer = lru_cache(consumer, nil, create_new_consumer, consumer)
```

Reply:
Yes，I‘ll refactor this code.